### ISSUE TITLE
twisted.test.test_strcred.TestFileDBChecker.setUp relies on GC to write contents of file used for testing

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3519]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3519|
|Type|defect|
|Created|2008-10-31 22:41:03Z|

This `setUp` opens a file and writes to it and then drops the reference.  If the file isn't garbage collected before the tests run, the bytes won't be there and the test will fail.
d

Attachments:

* [strcred-file.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/6c9/6c9d4c93d040a158767032bb8f21db2f3e713cb2/2ddbd59770bbf1a261f8070261106f1b439de99c.patch) (874 bytes) - added by exarkun on 2008-10-31 22:42:00Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3519 3519
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1225492863000000 1225492863000000
changetime__1225634370000000 1225634370000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

Attached fixes it by using `FilePath.setContent`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Please apply.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [25273]) Apply strcred-file.patch

Author: exarkun
Reviewer: therve
Fixes: [#3519](https://github.com/twisted/twisted/issues/3519)

Use `FilePath` in the strcred tests to write test data to a file.  A
useful consequence of this is that any obscure details about how files
should be written (such as that they should be closed explicitly) is
done for us.  This helps Python runtimes which don't do refcounting
by making sure the data actually ends up in the file before the tests
need it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
